src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -0,0 +1,94 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedWriter;
+import java.io.Closeable;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.LineIterator;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PersistableSlidingWindow extends SlidingWindow<SlidingWindowData> implements Closeable {
+  private static final Logger LOG = LogManager.getLogger(PersistableSlidingWindow.class);
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+
+  protected static final String SEPARATOR = "\n";
+
+  protected final long writePeriod;
+
+  private Path path;
+  private BufferedWriter writer;
+  private long timestamp = -1;
+
+  public PersistableSlidingWindow(int slidingWindowSize,
+                                  long writePeriod,
+                                  TimeUnit timeUnit,
+                                  String filePath) {
+    super(slidingWindowSize, timeUnit);
+    this.writePeriod = writePeriod;
+    try {
+      this.path = Paths.get(filePath);
+      if (Files.exists(path)) {
+        loadFromFile(path);
+      } else {
+        Files.createFile(path);
+      }
+      this.writer = new BufferedWriter(new FileWriter(filePath, false));
+    } catch (IOException e) {
+      LOG.error("Couldn't create file {} to perform young generation tuning", path, e);
+      throw new IllegalArgumentException("Couldn't create or read a file at " + filePath);
+    }
+  }
+
+  @Override
+  public void next(SlidingWindowData slidingWindowData) {
+    if (timestamp == -1) {
+      timestamp = slidingWindowData.getTimeStamp();
+    }
+    long currTimestamp = slidingWindowData.getTimeStamp();
+    super.next(slidingWindowData);
+    long timestampDiff = currTimestamp - timestamp;
+    if (timeUnit.convert(timestampDiff, TimeUnit.MILLISECONDS)
+        > timeUnit.convert(writePeriod, TimeUnit.MILLISECONDS)) {
```

Comment:
Why are we doing this? Are you trying to ensure at most one entry in queue per hour? If so, then shouldn't `super.next()` be called within this `if` ? and outside the if we should just be incrementing a count within that data point (if at all needed).

Reply:
The way I had this before, I was writing every write period, but not bucketizing the data.

I've changed this so that data is bucketized and writes are tied to bucket width (in units of time)